crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -711,15 +721,51 @@ impl AcirContext {
     pub(crate) fn brillig(
         &mut self,
         code: Vec<BrilligOpcode>,
-        inputs: Vec<AcirVar>,
-        output_len: usize,
-    ) -> Vec<AcirVar> {
-        let b_inputs =
-            vecmap(inputs, |i| BrilligInputs::Single(self.vars[&i].to_expression().into_owned()));
-        let outputs = vecmap(0..output_len, |_| self.acir_ir.next_witness_index());
-        let outputs_var =
-            vecmap(&outputs, |witness_index| self.add_data(AcirVarData::Witness(*witness_index)));
-        let b_outputs = vecmap(outputs, BrilligOutputs::Simple);
+        inputs: Vec<AcirValue>,
+        outputs: Vec<AcirType>,
+    ) -> Vec<AcirValue> {
+        let b_inputs = vecmap(inputs, |i| match i {
+            AcirValue::Var(var, _) => {
+                BrilligInputs::Single(self.vars[&var].to_expression().into_owned())
+            }
+            AcirValue::Array(vars) => {
+                let var_expressions = vars
+                    .iter()
+                    .map(|v| match v {
+                        AcirValue::Var(var, _) => self.vars[var].to_expression().into_owned(),
+                        // TODO: Support arrays of complex types (structures) as Brillig input
+                        _ => unreachable!("brillig does not supprot complex array inputs yet"),
+                    })
+                    .collect();
+                BrilligInputs::Array(var_expressions)
+            }
+        });
+
+        let mut b_outputs = Vec::new();
+        let outputs_var = vecmap(outputs, |output| {
+            match output {
+                AcirType::NumericType(_) => {
+                    let witness_index = self.acir_ir.next_witness_index();
+                    b_outputs.push(BrilligOutputs::Simple(witness_index));
+                    let var = self.add_data(AcirVarData::Witness(witness_index));
+                    AcirValue::Var(var, output.clone())
+                }
+                AcirType::Array(element_types, size) => {
+                    let mut witnesses = Vec::new();
+                    let mut array_values = im::Vector::new();
+                    for _ in 0..size {
+                        let witness_index = self.acir_ir.next_witness_index();
+                        witnesses.push(witness_index);
+                        let var = self.add_data(AcirVarData::Witness(witness_index));
+                        // TODO: Support arrays of complex types (structures) as Brillig output
+                        array_values.push_back(AcirValue::Var(var, element_types[0].clone()));
```

Comment:
```suggestion
                        for element_type in &element_types {
                            let witness_index = self.acir_ir.next_witness_index();
                            witnesses.push(witness_index);
                            let var = self.add_data(AcirVarData::Witness(witness_index));
                            array_values.push_back(AcirValue::Var(var, element_type.clone()));
                        }
```

Reply:
Done

Reply:
I imagine I could do the same thing above for inputs?

Reply:
Looks like it may be a tad more difficult for inputs if the BrilligInputs enum doesn't support struct elements.

Reply:
I already have this issue up so may leave it for a separate PR: https://github.com/noir-lang/noir/issues/1641

Reply:
Actually it is not too bad going to do it now

Reply:
Ok should work with struct inputs now too